presto-orc/src/main/java/com/facebook/presto/orc/reader/MapFlatSelectiveStreamReader.java

```diff
@@ -236,8 +236,8 @@ public int read(int offset, int[] positions, int positionCount)
 
         outputPositions = initializeOutputPositions(outputPositions, positions, positionCount);
 
-        if (presentStream != null && keyCount == 0) {
-            readAllNulls(positions, positionCount);
+        if (keyCount == 0 && presentStream == null) {
```

Comment:
Routing everything through the readNotAllNulls will work just fine too.

Reply:
Let us keep the readAllNulls usecase, there was performance testing done on that and some of the side tables have lot of nulls. So that path was optimized, so we should keep it as it is.

Reply:
Ok, I'll split current routing and include the `all nulls` case.